vortex-ffi/src/sink.rs

```diff
@@ -0,0 +1,80 @@
+use std::ptr::null_mut;
+
+use mpsc::Sender;
+use tokio::sync::mpsc;
+use tokio_stream::wrappers::ReceiverStream;
+use vortex::ArrayRef;
+use vortex::dtype::DType;
+use vortex::error::{VortexExpect, VortexResult, vortex_err};
+use vortex::stream::{ArrayStreamAdapter, ArrayStreamExt};
+
+use crate::array::vx_array;
+use crate::error::{try_or, vx_error};
+use crate::stream::{ArrayStreamInner, vx_array_stream};
+
+#[allow(non_camel_case_types)]
+/// The `sink` object is a writeable array stream, used to go from an external iterator of values
+/// into a `vx_array_stream`.
+pub struct vx_array_sink {
+    sink: Sender<VortexResult<ArrayRef>>,
+}
+
+#[repr(C)]
+#[allow(non_camel_case_types)]
```

Comment:
I think this can be file wide: `#![allow(non_camel_case_types)`. Then we don't have to repeat that for each type.

Reply:
I want to explicitly mark only public things